### DISCUSSION TITLE
Next.js 13 / Server Components

### DISCUSSION THREAD

**Discussion by nandorojo** [urql-graphql/urql#2775] [Q&A]
I wanted to ask about Next.js 13's server components. I know that many data fetching library authors are a bit stumped about how to support server components and what the right patterns are. 

On the server, it seems like we could call a plain `await executeQuery()` to get data. However, I'm not sure how this would then send data to the cache.

Nothing urgent here, just curious if you guys have any thoughts about it.

**Comment by JoviDeCroock** [comment]
When server-components execute on the server they won't send the data back down to the client as they aren't shared with the client as the client will only get a JSON-ified VDOM `diff`. I don't think currently there are any ways to send the data to the client as far as I know but yes using `useQuery` or `client.query().toPromise()` should just work.
